include/sas.h

```diff
@@ -397,8 +469,22 @@ class SAS
 
   static sas_log_callback_t* _log_callback;
 
-private:
+  /// Converts the format of a log raised within the SAS-Client to that expected by
+  /// the common log callback, and calls the common log callback with the converted
+  /// arguments.
+  static void sasclient_log_callback(log_level_t level,
+                                     const char *module,
+                                     int line_number,
+                                     const char *fmt,
+                                     ...);
```

Comment:
Why does this need to be public.

Reply:
To allow the SAS_LOG macro in sas_internal.h access to it.

Reply:
Macros don't care about access permissions. They are expanded before the C++ compiler even gets run.

Reply:
Hmmm my compiler is throwing wobbly's I don't understand for now then, it explicitly complains at me if this is private, and is only called by that macro.